pkg/model_user_in.go

```diff
@@ -28,3 +35,33 @@ type UserIn struct {
 
 	Password string `json:"password"`
 }
+
+// UnmarshalJSON overrides default Unmarshal method to verify JSON fields 
+func (userIn *UserIn) UnmarshalJSON(data []byte) error {
+	type UserIn2 UserIn
+	var userIn2 UserIn2
+	if err := json.Unmarshal(data, &userIn2); err != nil {
+		return err
+	}
+	
+	value := reflect.ValueOf(userIn2)
+	for i := 0; i < value.NumField(); i++ {
+		fieldName := value.Type().Field(i).Name
+		fieldValue := value.Field(i).Interface()
+		if fieldValue == "" {
+			return fmt.Errorf("Missing or empty field '%+v' for UserIn", fieldName)
+		} else if fieldName == "Password" {
+			if len(fieldValue.(string)) < 6 {
+				return fmt.Errorf("Password must be 6 or more characters")
+			}
```

Comment:
Might be good to factor this out to make it easier to apply more or different rules down the track.